Move concepts and cloud pages to guides and reorganize (#10328)
PrefectHQ/prefect: docs/guides/index.md

tags:
    - guides
    - how to
hide:
  - toc
search:
  boost: 2
---

# Guides

This section of the documentation contains guides for common workflows and use cases. 

| Keyword                                                  | Description                                                                                        |
| -------------------------------------------------------- | -------------------------------------------------------------------------------------------------- |
| [Storage](/guides/deployment/storage-guide/) | Store your flow code for deployment runs. |
| [Docker](/guides/deployment/docker/) | Deploy workflows with Docker containers. |
| [Kubernetes](/guides/deployment/helm-worker/) | Deploy flows on Kubernetes. |

> | [Push Work Pools](/guides/deployment/push-work-pools/) |  Execute serverless flows on ECS, ACI, or Cloud Run without running a worker. | 

Reviewer: ```suggestion
| [Push Work Pools](/guides/deployment/push-work-pools/) |  Execute flows on serverless infrastructure like AWS ECS, Azure Container Instances, or Google Cloud Run without a worker. | 
```

| [ECS](https://prefecthq.github.io/prefect-aws/#using-prefect-with-aws-ecs) |  Run flows on AWS ECS. |
| [Azure Container Instances](/guides/deployment/aci/) |  Deploy workflows with ACI. |
| [Custom Workers](/guides/deployment/developing-a-new-worker-type/) | Develop your own worker type. | 
| [Profiles & Settings](/guides/settings/) | Configure Prefect and save your settings. |

> | [Logging](/guides/logs/) | Get the information you need. |

Reviewer: ```suggestion
| [Logging](/guides/logs/) | Configure Prefect's logger and aggregate logs from other tools. |
```

> | [Testing](/guides/testing/) | Easily test your workflows. |

Reviewer: ```suggestion
| [Unit Testing](/guides/testing/) | Use Prefect's test harness to test your flows. |
```

> | [Troubleshooting](/guides/troubleshooting/) | For the times everything doesn't go according to plan. |

Reviewer: ```suggestion
| [Troubleshooting](/guides/troubleshooting/) | Identify and resolve common issues with Prefect. |
```

| [Runtime context](/guides/runtime-context/) | Use information about currently running flows in your flow code.  |

> | [Variables](/guides/variables/) | Save small bits of configuration. | 

Reviewer: ```suggestion
| [Variables](/guides/variables/) | Store and retrieve configuration data. | 
```

> | [Runtime context](/guides/runtime-context/) | Use information about currently running flows in your flow code.  |

Reviewer: ```suggestion
| [Runtime context](/guides/runtime-context/) | Enable a flow to access metadata about itself and its context when it runs.  |
```

| [Webhooks](/guides/webhooks/) | Receive, observe, and react to events from other systems. |
| [State Change Hooks](/guides/state-change-hooks/) | Execute code in response to state changes. |

> | [Dask and Ray](/guides/dask-ray-task-runners/) | Parallelize your code with task runners. |

Reviewer: ```suggestion
| [Dask and Ray](/guides/dask-ray-task-runners/) | Scale your flows with parallel computing frameworks. |
```

> | [Dask and Ray](/guides/dask-ray-task-runners/) | Scale your flows with parallel computing frameworks like Dask and Ray. |

Reviewer: ```suggestion
| [Dask and Ray](/guides/dask-ray-task-runners/) | Scale your flows with parallel computing frameworks. |
```

| [Hosting](/guides/host/) | Host your own Prefect Server. |

> | [Migration Guide](/guides/migration-guide/) | Learn how to migrate from Prefect 1 to Prefect 2. |

Reviewer: ```suggestion
| [Migration Guide](/guides/migration-guide/) | Migrate from Prefect 1 to Prefect 2. |
```

> | [Prefect Recipes](../recipes/recipes/) |  See common workflows. |

Reviewer: ```suggestion
| [Prefect Recipes](../recipes/recipes/) |  Common, extensible examples for setting up Prefect. |
```
